ibm/service/cos/resource_ibm_cos_bucket_lifecycle_configuration.go

```diff
@@ -0,0 +1,568 @@
+	} else {
+		return nil
+	}
+}
+
+func lifecycleExpirationSet(expiration []interface{}) *s3.LifecycleExpiration {
+	if len(expiration) == 0 {
+		return nil
+	}
+
+	var result *s3.LifecycleExpiration
+
+	resultValue := s3.LifecycleExpiration{}
+
+	if expiration[0] == nil {
+		return result
+	}
+
+	expirationMap := expiration[0].(map[string]interface{})
+	if v, ok := expirationMap["date"].(string); ok && v != "" {
+		t, _ := time.Parse(time.RFC3339, v)
+		resultValue.Date = aws.Time(t)
+	}
+
+	if v, ok := expirationMap["days"]; ok {
+		daysSet := int64(v.(int))
+		if daysSet > 0 {
+			days := daysSet
+			resultValue.Days = aws.Int64(days)
+		}
+	}
+
+	// This cannot be specified with Days or Date
+	if v, exist := expirationMap["expired_object_delete_marker"]; exist && resultValue.Date == nil && resultValue.Days == nil {
+		resultValue.ExpiredObjectDeleteMarker = aws.Bool(v.(bool))
+	}
+
+	result = &resultValue
+
+	return result
+}
+
+func lifecycleRuleFilterSet(filter []interface{}) *s3.LifecycleRuleFilter {
+	var result *s3.LifecycleRuleFilter
+	resultValue := s3.LifecycleRuleFilter{}
+	if filter[0] == nil {
+		resultValue.Prefix = aws.String("")
+		result = &resultValue
+		return result
+	}
+
+	filterMap := filter[0].(map[string]interface{})
+	if v, ok := filterMap["prefix"].(string); ok {
+		resultValue.Prefix = aws.String(v)
+	}
+	result = &resultValue
+	return result
+}
+
+func noncurrentVersionExpirationSet(noncurrentVersionExpiration map[string]interface{}) *s3.NoncurrentVersionExpiration {
+
+	var result *s3.NoncurrentVersionExpiration
+	resultValue := s3.NoncurrentVersionExpiration{}
+	if v, ok := noncurrentVersionExpiration["noncurrent_days"]; ok {
+		resultValue.NoncurrentDays = aws.Int64(int64(v.(int)))
+	}
+	result = &resultValue
+	return result
+}
+
+func transitionsSet(transitions []interface{}) []*s3.Transition {
+	if len(transitions) == 0 {
+		return nil
+	}
+
+	var results []*s3.Transition
+	var transition = s3.Transition{}
+	if transitions[0] == nil {
+		return results
+	}
+
+	transitionMap := transitions[0].(map[string]interface{})
+	if v, ok := transitionMap["date"].(string); ok && v != "" {
+		t, _ := time.Parse(time.RFC3339, v)
+		transition.Date = aws.Time(t)
+	}
+
+	// Only one of "date" and "days" can be configured
+	// so only set the transition.Days value when transition.Date is nil
+	// By default, transitionMap["days"] = 0 if not explicitly configured in terraform.
+	if v, ok := transitionMap["days"]; ok && v.(int) >= 0 && transition.Date == nil {
+		transition.Days = aws.Int64(int64(v.(int)))
+	}
+
+	if v, ok := transitionMap["storage_class"].(string); ok && v != "" {
+		transition.StorageClass = aws.String(v)
+	}
+
+	results = append(results, &transition)
+
+	return results
+}
+
+func lifecycleConfigurationSet(lifecycleConfigurationList []interface{}) []*s3.LifecycleRule {
+
+	var lifecycleRules []*s3.LifecycleRule
+
+	for _, lifecycleRuleMapRaw := range lifecycleConfigurationList {
```

Comment:
since we're not typecasting lifecycleConfigurationList elements can be of type empty interface we can also introduce a nil check before trying type assertion on lifecycleRuleMapRaw. 
such as : 
```
if lifecycleRuleMapRaw == nil {
    continue
}
```

Reply:
Yes some of them can be empty , but status and id are required parameters so they will always not nil.